src/index/IndexImpl.cpp

```diff
@@ -363,44 +388,54 @@ void IndexImpl::createFromFiles(
       ad_utility::uniqueBlockView(firstSorter.getSortedOutput());
+  auto firstSorterWithUniqueIR{fromGenerator(std::move(firstSorterWithUnique))};
 
   if (!loadAllPermutations_) {
     createInternalPsoAndPosAndSetMetadata();
     // Only two permutations, no patterns, in this case the `firstSorter` is a
     // PSO sorter, and `createPermutationPair` creates PSO/POS permutations.
     createFirstPermutationPair(NumColumnsIndexBuilding,
-                               std::move(firstSorterWithUnique));
+                               std::move(firstSorterWithUniqueIR));
     configurationJson_["has-all-permutations"] = false;
   } else if (!usePatterns_) {
     createInternalPsoAndPosAndSetMetadata();
     // Without patterns, we explicitly have to pass in the next sorters to all
     // permutation creating functions.
     auto secondSorter = makeSorter<SecondPermutation>("second");
     createFirstPermutationPair(NumColumnsIndexBuilding,
-                               std::move(firstSorterWithUnique), secondSorter);
+                               std::move(firstSorterWithUniqueIR),
+                               secondSorter);
     firstSorter.clearUnderlying();
 
     auto thirdSorter = makeSorter<ThirdPermutation>("third");
+    auto secondSorterBlocksG{secondSorter.getSortedBlocks<0>()};
+    auto secondSorterBlocks{fromGenerator(std::move(secondSorterBlocksG))};
     createSecondPermutationPair(NumColumnsIndexBuilding,
-                                secondSorter.getSortedBlocks<0>(), thirdSorter);
+                                std::move(secondSorterBlocks), thirdSorter);
     secondSorter.clear();
+
+    auto thirdSorterBlocksG{thirdSorter.getSortedBlocks<0>()};
+    auto thirdSorterBlocks{fromGenerator(std::move(thirdSorterBlocksG))};
     createThirdPermutationPair(NumColumnsIndexBuilding,
-                               thirdSorter.getSortedBlocks<0>());
+                               std::move(thirdSorterBlocks));
     configurationJson_["has-all-permutations"] = true;
   } else {
     // Load all permutations and also load the patterns. In this case the
     // `createFirstPermutationPair` function returns the next sorter, already
     // enriched with the patterns of the subjects in the triple.
     auto patternOutput = createFirstPermutationPair(
-        NumColumnsIndexBuilding, std::move(firstSorterWithUnique));
+        NumColumnsIndexBuilding, std::move(firstSorterWithUniqueIR));
     firstSorter.clearUnderlying();
     auto thirdSorterPtr =
         buildOspWithPatterns(std::move(patternOutput.value()),
                              *indexBuilderData.sorter_.internalTriplesPso_);
     createInternalPsoAndPosAndSetMetadata();
-    createThirdPermutationPair(NumColumnsIndexBuilding + 2,
 
-                               thirdSorterPtr->template getSortedBlocks<0>());
+    auto thirdSorterBlocksG{thirdSorterPtr->template getSortedBlocks<0>()};
+    auto thirdSorterBlocks{fromGenerator(std::move(thirdSorterBlocksG))};
```

Comment:
see above, no need for two separate variables with dedicated names each.

Reply:
fromGenerator removed [here](https://github.com/ad-freiburg/qlever/pull/2151/commits/422819e4030ddc2d1f2111ee3145b76a9cefbebc)